internal/cache/controller_obj_cache.go

```diff
@@ -2700,25 +2700,15 @@ func checkAndSetCloudType(client *clients.AviClient) bool {
 func checkPublicCloud(client *clients.AviClient) bool {
 	if lib.IsPublicCloud() {
 		// Handle all public cloud validations here
-		networkName := lib.GetNetworkName()
-		if networkName == "" && lib.GetCloudType() != lib.CLOUD_GCP {
-			if lib.GetCloudType() == lib.CLOUD_AWS {
-				//check for vipNetworkList if networkName not present for AWS Clouds
-				utils.AviLog.Infof("networkName not specified, checking for vipNetworkList as cloud is AWS")
-				vipNetworkList, err := lib.GetVipNetworkList()
-				if err == nil && len(vipNetworkList) != 0 {
-					utils.AviLog.Infof("Proceeding as multi-vip enabled")
-					return true
-				} else {
-					utils.AviLog.Errorf("vipNetworkList not specified, syncing will be disabled.")
-					if err != nil {
-						utils.AviLog.Errorf("%s", err.Error())
-					}
-					return false
-				}
+		if lib.GetCloudType() != lib.CLOUD_GCP {
```

Comment:
Do you know why we make an exception in case of GCP?

Reply:
Good point. I had followed the existing logic of ignoring GCP cloud. 
@parimanur can you please review this PR  .. specifically here if we need the  check for GCP too ?

Reply:
@sudswasavi @monotosh-avi  In GCP we can accept ipamdnsproviderprofile when creating the VIP. We can change the condition for IsPublicCloud to specifically check for AWS and Azure.

Reply:
@parimanur I saw that the function checkPublicCloud() is being called only from ValidateUserInput(), In that case are we fine if you remove the GCP cloud check ? 
I didn't understand the ipamdnsproviderprofile part ..

Reply:
@monotosh-avi Yes, you can delete the check. I was talking about in general where the behaviour of GCP vs other public clouds.

Reply:
Got it. Done